packages/dd-trace/src/config.js

```diff
@@ -185,73 +136,12 @@ class Config {
         process.env.DD_SERVICE_MAPPING.split(',').map(x => x.trim().split(':'))
       ) : {}
     )
-    const DD_TRACE_STARTUP_LOGS = coalesce(
-      options.startupLogs,
-      process.env.DD_TRACE_STARTUP_LOGS,
-      false
-    )
-
-    const DD_OPENAI_LOGS_ENABLED = coalesce(
-      options.openAiLogsEnabled,
-      process.env.DD_OPENAI_LOGS_ENABLED,
-      false
-    )
 
     const DD_API_KEY = coalesce(
       process.env.DATADOG_API_KEY,
       process.env.DD_API_KEY
     )
 
-    const inAWSLambda = process.env.AWS_LAMBDA_FUNCTION_NAME !== undefined
-
-    const isGCPFunction = getIsGCPFunction()
-    const isAzureFunctionConsumptionPlan = getIsAzureFunctionConsumptionPlan()
-
-    const inServerlessEnvironment = inAWSLambda || isGCPFunction || isAzureFunctionConsumptionPlan
-
-    const DD_INSTRUMENTATION_TELEMETRY_ENABLED = coalesce(
-      process.env.DD_TRACE_TELEMETRY_ENABLED, // for backward compatibility
-      process.env.DD_INSTRUMENTATION_TELEMETRY_ENABLED, // to comply with instrumentation telemetry specs
-      !inServerlessEnvironment
```

Comment:
ℹ️ after merging https://github.com/DataDog/dd-trace-js/pull/4098/files#diff-35b625667f1fbe3cfe5d2d1290484d9643c044d90ece6dea383d9e26a1633755R217 this also takes into account whether you're in a jest worker